### PR TITLE
Improve ssh analytics

### PR DESCRIPTION
**Pull request by jeanp413**
How to test for failure:
- SSH gateway:
  - To force the connection to fail we can remove all our ssh keys so you'll be required to use password authentication and then just type an incorrect password
- Local app:
  - ....

### CODE REVIEW

**Comment by akosyakov** [comment]
@jeanp413 It would be good to specify test cases 🙏 there are quite some cases, i.e. how do I check local app vs gateway, and then for both failure vs success?

**Comment by jeanp413** [comment]
Added steps to test using gateway, not sure how to make a test for local app maybe just throw an error randomly :thinking:

**Comment by akosyakov** [comment]
Do you test it again IDE lab project in Mixpanel?

**Comment by andreafalzetti** [comment]
What do I need to do for testing this? Start a new workspace? 🤔

**Comment by akosyakov** [comment]
> That the event is correctly send there? yeah or do you mean create a funnel?

@jeanp413 No, more like share with @andreafalzetti how he can verify it. A link in the PR description to some notion page would help.

**Comment by akosyakov** [comment]
> check that the analytics event is fired successfully and only when the ssh connection is done by the remote-ssh extension. You need to clone this repo and hit f5 like any js project and then open a workspace and open in vscode desktop windows that is running the extension, if you have it installed delete it first so the dev version is used

We should document it somewhere and link from PR descriptions, like in CONTRIBUTING.md

**Comment by jeanp413** [comment]
[Done](https://github.com/gitpod-io/gitpod-vscode-desktop/blob/master/docs/CONTRIBUTING.md)

File: src/remoteConnector.ts [hunk 1]
```diff
@@ -766,4 +760,53 @@ export default class RemoteConnector extends Disposable {
...
+			const [, sshDestStr] = remoteUri.authority.split('+');
+			const sshDest: { user: string; hostName: string } = JSON.parse(Buffer.from(sshDestStr, 'hex').toString('utf8'));
+			const gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
+			const host = new URL(gitpodHost).host;
+			const connectionSuccessful = sshDest.hostName.endsWith(host) && await isRemoteExtensionHostRunning();
+
+			const connectionInfo = this.context.globalState.get<SSHConnectionParams & { usingSSHGateway: boolean }>(sshDestStr);
+			if (connectionInfo) {
+				const kind = connectionInfo.usingSSHGateway ? 'gateway' : 'local-app';
+				if (connectionSuccessful) {
```
Comment (andreafalzetti) [inline src/remoteConnector.ts:786]: do we consider successful a connection even if the user failed to enter the right password?

Comment (jeanp413) [reply src/remoteConnector.ts:786]: Nope, it should send the `failed` status

**Review by andreafalzetti — APPROVED** [review]
tested for ssh gateway and works as advertised

**Pull request merged**